frontend/public/kubevirt/components/utils/lazyRenderer.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import * as _ from 'lodash-es';
+import { inject } from '../../../components/utils';
+
+const DEFAULT_THROTTLE_DELAY = 250; // in ms
+
+/**
+ * Avoids eager re-rendering for frequent redux store changes.
+ *
+ * Assumption: re-render requests are frequent, so skipping a few of them
+ * has no impact on rendered information from user's perspective.
+ *
+ * If that assumption is not met, the use of this component is not recommended.
+ */
+export class LazyRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isRenderingEnabled: false,
+    };
+    this.enableRendering = _.throttle(() => {
```

Comment:
just curios, isn't it better to limit redux change rate in the first place? (e.g. have a throttle on the action)

Reply:
It would be definitely better but the "actions" are driven by k8s API as they are fired on object changes. I don't know how to throttle the producer in that case. We can batch changes (i.e. round the `Firehose` component) before storing them to redux, but its way more complicated.

The listener on our side is implemented by react-redux. It would be perfect place to implement throttling as well but I failed to find an option for that. Patching react-redux does not serve our needs due to time constraints. This `LazyReducer` does the expected thing on a few lines.

I hope I am not wrong, but I do not expect the `LazyRouter` to be used heavily throughout the code. Its use case is
- frequent relevant changes of redux
- complex component structure or expensive renderers (like charts, etc)
- partially: expensive `shouldComponentUpdate` or to much effort to implement them consistently

The last option can be "tweaked" by using `PureCompoent` is specific cases.

With `LazyRenderer` approach, there is no lost of information from user's POV.